main.tf

```
@@ -142,6 +142,24 @@ resource "aws_s3_bucket" "default" {
     }
   }
 
+  dynamic "replication_configuration" {
+    for_each = var.s3_replication_enabled ? toset([var.s3_replica_bucket_arn]) : []
+    content {
+      role = aws_iam_role.replication[0].arn
```

```suggestion
      role = join("", aws_iam_role.replication.*.arn)
```

`[0[` will fail when `count=0`.
Also `role = try(aws_iam_role.replication[0].arn, "")` could be used

---

@aknysh so I actually tried it with count=0 and it didn't have any issues -- unless maybe it's a version-specific issue? Either that or it doesn't error because it's already wrapped inside of a dynamic block.

---

And FWIW, all of the automated tests here run with count=0, and they passed.